packages/gatsby-source-nacelle/src/source-nodes/contentful-preview.js

```diff
@@ -0,0 +1,53 @@
+const { nacelleClient } = require('../services');
+const { replaceKeys } = require('../utils');
+
+module.exports = async function (gatsbyApi, pluginOptions) {
+  const { actions, createNodeId, createContentDigest } = gatsbyApi;
+  const { createNode } = actions;
+
+  const {
+    nacelleSpaceId,
+    nacelleGraphqlToken,
+    contentfulPreviewSpaceId,
+    contentfulPreviewApiToken
+  } = pluginOptions;
+
+  const client = nacelleClient({
+    previewMode: true,
+    nacelleSpaceId,
+    nacelleGraphqlToken,
+    contentfulPreviewSpaceId,
+    contentfulPreviewApiToken
+  });
+
+  try {
+    const contentData = await client.data
+      .allContent()
+      .then((
+        content // change name of reserved keys
+      ) => replaceKeys(content, { id: 'remoteId', fields: 'remoteFields' }))
+      .catch((err) => {
+        throw new Error(
+          `Problem fetching content:\n\n${JSON.stringify(err, null, 2)}`
```

Comment:
Won't this error just get caught by the `try... catch`? If so, we'd end up with `Problem sourcing content from Contentful Preview API: [object object]` or something of that sort. What about something like:

```js
const contentData = await client.data.allContent();
const formattedData = replaceKeys(content, { id: 'remoteId', fields: 'remoteFields' });

contentData.forEach(...)
```

```js
...
} catch (err) {
  console.error(`Problem sourcing content from Contentful Preview API: ${err.message}`);
}
```
```

Reply:
Right! Changed :)